examples/3d_examples/OctopusSwimming/octopus_initializer_functions.py

```diff
@@ -0,0 +1,147 @@
+        ).reshape(3, 3)
+        direction_from_center_to_rod = rotation_matrix @ binormal
+
+        angle_wrt_center_rod.append(angle_btw_straight_rods * i)
+
+        # Compute the rotation matrix, for getting the correct banked angle.
+        normal_banked_rod = rotation_matrix @ normal
+        # Rotate direction vector around new normal to get the new direction vector.
+        # Note that we are doing ccw rotation and direction should be towards the center.
+        rotation_matrix_banked_rod = _get_rotation_matrix(
+            (np.pi / 2 - bank_angle), normal_banked_rod.reshape(3, 1)
+        ).reshape(3, 3)
+        direction_banked_rod = rotation_matrix_banked_rod @ direction
+
+        start_rod = (
+            start
+            + (direction_from_center_to_rod)
+            * (
+                # center rod            # this rod
+                +2 * base_radius
+                + base_radius
+            )
+            * 0
+        )
+
+        radius = np.linspace(base_radius, base_radius / taper_ratio, n_elems + 1)
+        radius_mean = (radius[:-1] + radius[1:]) / 2
+
+        rod = ea.CosseratRod.straight_rod(
+            n_elements=n_elems,
+            start=start_rod,
+            direction=direction_banked_rod,
+            normal=normal_banked_rod,
+            base_length=base_length,
+            base_radius=radius_mean.copy(),
+            density=rho_s,
+            nu=0.0,  # internal damping constant, deprecated in v0.3.0
+            youngs_modulus=youngs_modulus,
+            shear_modulus=shear_modulus,
+        )
+
+        rod_list.append(rod)
+        arm_rod_list.append(rod)
+
+    # Octopus head initialization
+    slenderness_ratio_head = 5.787
+
+    octopus_head_length = 2 * base_radius * slenderness_ratio_head / 2
+    octopus_head_n_elems = int(n_elems * octopus_head_length / base_length)
+
+    octopus_head_radius = (
+        2 * base_radius * np.linspace(0.9, 1.0, octopus_head_n_elems) ** 3
+    )
+
+    body_rod = ea.CosseratRod.straight_rod(
+        n_elements=octopus_head_n_elems,
+        start=start,
+        direction=-direction,
+        normal=normal,
+        base_length=octopus_head_length,
+        base_radius=octopus_head_radius,  # .copy(),
+        density=rho_s,
+        nu=0.0,  # internal damping constant, deprecated in v0.3.0
+        youngs_modulus=youngs_modulus,
+        shear_modulus=shear_modulus,
+    )
+    rod_list.append(body_rod)
+
+    rigid_body_list = []
+    sphere_com = body_rod.position_collection[..., -1]
+    sphere_diameter = np.max(octopus_head_radius)
+    sphere = ea.Sphere(center=sphere_com, base_radius=sphere_diameter, density=rho_s)
+    rigid_body_list.append(sphere)
+
+    return arm_rod_list, rod_list, rigid_body_list
+
+
+def initialize_activation_functions(
+    env,
+    activation_period,
+    activation_level_max,
+    wave_number=0.05,
+    start_non_dim_length=0.0,
+    end_non_dim_length=1.0,
```

Comment:
would be helpful here too, specially for `env`

Reply:
5de8a92